elements/lisk-chain/src/data_access/data_access.ts

```diff
@@ -384,7 +384,7 @@ export class DataAccess {
 	public deserializeTransaction(
 		transactionJSON: TransactionJSON,
 	): BaseTransaction {
-		return this._transactionAdapter.fromJSON(transactionJSON);
+		return this._transactionAdapter.decode(transactionJSON);
```

Comment:
Maybe rename `transactionJSON` to `transactionBuffer` as `decode()` deals with binary data and not JSON?

Reply:
same